setup.py

```diff
@@ -89,7 +89,7 @@ def pypi_link(pkg_filename):
 
 setup_params = dict(
     name="setuptools",
-    version="40.8.0",
```

Comment:
Although you _could_ use `get_version` here, this usage would be discouraged. Better would be to store the value in setup.cfg under `metadata.version` (even if in duplicate), although doing so leads to [a limitation in bump2version](https://github.com/c4urself/bump2version/issues/62). What's more concerning about this change is the issues it introduces if setuptools were to move to a declarative config (something that's definitely desirable)... and this speaks to the resistance to accepting this change--it's encouraging patterns that run contrary to the best practices.

Reply:
So what is the ETA to introduce the best practices to `pip`? Why doesn't it use best practices?

Reply:
Support for declarative config is relatively recent and I haven't spent much time evangelizing it, so it's probably no big surprise that some projects haven't adopted it. Even setuptools itself, which because it installs itself, would have had no minimum dependency, only adopted declarative config many months after it was available. It's just a matter of priorities and resources. Perhaps that team would accept a pull request to use declarative config; I don't know.